mongo/integration/mtest/mongotest.go

```diff
@@ -696,9 +698,14 @@ func verifyServerParametersConstraints(serverParameters map[string]bson.RawValue
 	return nil
 }
 
-func verifyAuthConstraint(expected *bool) error {
-	if expected != nil && *expected != testContext.authEnabled {
-		return fmt.Errorf("test requires auth value: %v, cluster auth value: %v", *expected, testContext.authEnabled)
+func verifyAuthConstraint(expectedAuth *bool, expectedAuthEnabled *bool) error {
```

Comment:
Nit:

IIUC:
1. `verifyAuthConstraint` is called when running both unified spec tests and non-unified spec tests through `verifyRunOnBlockConstraint`.
2. `verifyAuthConstraint` is called from `verifyConstraints` if an `mtest.T` instance is created with the `Auth` option ([example](https://github.com/mongodb/mongo-go-driver/blob/ccd27717f6ed6ba4f77b3213df84c1f7bc3eb630/mongo/integration/client_test.go#L82:L82)).

Would it be preferable if the caller of `verifyAuthConstraint` passes one `*bool`? (1) can determine whether to pass `Auth` or `AuthEnabled` from the parsed `RunOnBlock` depending on which is set.